file_store/src/mobile_session.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    error::DecodeError,
+    traits::{MsgDecode, MsgTimestamp, TimestampDecode},
+    Error, Result,
+};
+use chrono::{DateTime, Utc};
+use helium_crypto::PublicKeyBinary;
+use helium_proto::services::poc_mobile::{
+    DataTransferRadioAccessTechnology, DataTransferSessionIngestReportV1, DataTransferSessionReqV1,
+};
+use serde::Serialize;
+
+impl MsgTimestamp<Result<DateTime<Utc>>> for DataTransferSessionIngestReportV1 {
+    fn timestamp(&self) -> Result<DateTime<Utc>> {
+        self.received_timestamp.to_timestamp_millis()
+    }
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct DataTransferSessionIngestReport {
+    pub received_timestamp: DateTime<Utc>,
+    pub report: DataTransferSessionReq,
+}
+
+impl MsgDecode for DataTransferSessionIngestReport {
+    type Msg = DataTransferSessionIngestReportV1;
+}
+
+impl TryFrom<DataTransferSessionIngestReportV1> for DataTransferSessionIngestReport {
+    type Error = Error;
+
+    fn try_from(v: DataTransferSessionIngestReportV1) -> Result<Self> {
+        Ok(Self {
+            received_timestamp: v.timestamp()?,
+            report: v
+                .report
+                .ok_or_else(|| Error::not_found("data transfer session report"))?
+                .try_into()?,
+        })
+    }
+}
+
+#[derive(Serialize, Clone, Debug)]
+pub struct DataTransferSessionReq {
+    pub pub_key: PublicKeyBinary,
+    pub upload_bytes: u64,
+    pub download_bytes: u64,
+    pub radio_access_technology: DataTransferRadioAccessTechnology,
+    pub session_id: String,
+    pub payer: PublicKeyBinary,
+    pub imsi_hash: Vec<u8>,
+    pub timestamp: DateTime<Utc>,
+    pub signature: Vec<u8>,
+}
+
+impl MsgTimestamp<Result<DateTime<Utc>>> for DataTransferSessionReqV1 {
+    fn timestamp(&self) -> Result<DateTime<Utc>> {
+        self.timestamp.to_timestamp_millis()
+    }
+}
+
+impl TryFrom<DataTransferSessionReqV1> for DataTransferSessionReq {
+    type Error = Error;
+
+    fn try_from(v: DataTransferSessionReqV1) -> Result<Self> {
+        let timestamp = v.timestamp()?;
+        let radio_access_technology = DataTransferRadioAccessTechnology::from_i32(
+            v.radio_access_technology,
+        )
```

Comment:
```suggestion
        let radio_access_technology = v.radio_access_technology(),
```
would this not work in this scenario?